<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>966</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Phase scan </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="read_monitor_pushButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>220</y>
      <width>161</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Plot monitor devices</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>491</width>
      <height>241</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>-1</x>
       <y>-1</y>
       <width>491</width>
       <height>241</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="phases_verticalLayout"/>
    </widget>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>320</y>
      <width>356</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="1">
      <widget class="QSpinBox" name="numevents_spinBox">
       <property name="maximum">
        <number>10000</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QComboBox" name="event_comboBox"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_1">
       <property name="text">
        <string>Event</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Measurements</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="plot_phases_pushButton">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>30</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Plot phases</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>280</y>
      <width>190</width>
      <height>304</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QPushButton" name="selectall_pushButton">
       <property name="text">
        <string>Select all</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="clearall_pushButton">
       <property name="text">
        <string>Clear all</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Phase set</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Delta (deg)</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QCheckBox" name="checkBox_1">
       <property name="text">
        <string>RFQ</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_1">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QCheckBox" name="checkBox_2">
       <property name="text">
        <string>RFB</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QCheckBox" name="checkBox_3">
       <property name="text">
        <string>Tank 1</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_3">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QCheckBox" name="checkBox_4">
       <property name="text">
        <string>Tank 2</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_4">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QCheckBox" name="checkBox_5">
       <property name="text">
        <string>Tank 3</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QCheckBox" name="checkBox_6">
       <property name="text">
        <string>Tank 4</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QCheckBox" name="checkBox_7">
       <property name="text">
        <string>Tank 5</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_7">
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QListWidget" name="listWidget">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>100</y>
      <width>151</width>
      <height>111</height>
     </rect>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::MultiSelection</enum>
    </property>
    <item>
     <property name="text">
      <string>L:TO1IN</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>L:TO3IN</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>L:TO5OUT</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>L:D7TOR</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>L:D7LMSM</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>G:LINEFF</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>70</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Select monitor devices</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>430</y>
      <width>651</width>
      <height>141</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetFixedSize</enum>
     </property>
     <item row="0" column="1">
      <widget class="QPushButton" name="displayList_pushButton">
       <property name="text">
        <string>Display list</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QPushButton" name="genList_pushButton">
       <property name="text">
        <string>Generate ramp list</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="writeList_pushButton">
       <property name="text">
        <string>Write csv</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="scan_pushButton">
       <property name="text">
        <string>Scan devices</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="displayScan_pushButton">
       <property name="text">
        <string>Display data</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="writeScan_pushButton">
       <property name="text">
        <string>Write csv</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QPlainTextEdit" name="scan_plainTextEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>30</height>
        </size>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QPlainTextEdit" name="list_plainTextEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>30</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>410</y>
      <width>136</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Output filename:</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>966</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>plot_phases_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>update_plot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>305</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>565</x>
     <y>409</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selectall_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_all()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearall_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clear_all()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>genList_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generate_ramp_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>436</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>displayList_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>display_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>574</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>writeList_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>write_list()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>691</x>
     <y>483</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scan_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>scan()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>displayScan_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>display_scan_results()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>writeScan_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>write_scan_results()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>700</x>
     <y>554</y>
    </hint>
    <hint type="destinationlabel">
     <x>482</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>update_plot()</slot>
  <slot>add_param()</slot>
  <slot>select_all()</slot>
  <slot>clear_all()</slot>
  <slot>generate_ramp_list()</slot>
  <slot>display_list()</slot>
  <slot>write_list()</slot>
  <slot>scan()</slot>
  <slot>write_scan_results()</slot>
  <slot>display_scan_results()</slot>
 </slots>
</ui>
